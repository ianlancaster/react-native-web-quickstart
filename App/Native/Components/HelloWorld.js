import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';

export default class HelloWorld extends Component {
  render() {
    const style = StyleSheet.create({
      helloWorld: {
        color: 'black',
        textAlign: 'center',
      }
    });

    return (
      <View>
        <Text style={style.helloWorld}>Hello World</Text>
        <Text style={style.helloWorld}>Hello World</Text>
        <Text style={style.helloWorld}>Hello World</Text>
        <Text style={style.helloWorld}>Hello World</Text>
        <Text style={style.helloWorld}>Hello World</Text>
        <Text style={style.helloWorld}>Hello World</Text>
        <Text style={style.helloWorld}>Hello World</Text>
        <Text style={style.helloWorld}>Hello World</Text>
        <Text style={style.helloWorld}>Hello World</Text>
        <Text style={style.helloWorld}>Hello World</Text>
        <Text style={style.helloWorld}>Hello World</Text>
        <Text style={style.helloWorld}>Hello World</Text>
        <Text style={style.helloWorld}>Hello World</Text>
      </View>
    );
  }
}
