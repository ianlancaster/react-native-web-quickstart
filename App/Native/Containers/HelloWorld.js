import React, { Component, PropTypes } from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { connect } from 'react-redux'
import { toggleColor } from '../../Modules/helloWorld'

class HelloWorld extends Component {
  render () {
    const { toggleColor, color } = this.props
    const style = StyleSheet.create({
      mainView: {
        flexDirection: 'column',
        flex: 1,
        justifyContent: 'center'
      },
      helloWorld: {
        color: color,
        textAlign: 'center'
      }
    })

    return (
      <View style={style.mainView}>
        <Text onPress={toggleColor} style={style.helloWorld}>Hello World</Text>
      </View>
    )
  }
}

HelloWorld.propTypes = {
  color: PropTypes.string.isRequired,
  toggleColor: PropTypes.func.isRequired
}

const mapDispatchToProps = {
  toggleColor
}

const mapStateToProps = (state, ownProps) => ({
  color: state.helloWorld.color
})

export default connect(mapStateToProps, mapDispatchToProps)(HelloWorld)
