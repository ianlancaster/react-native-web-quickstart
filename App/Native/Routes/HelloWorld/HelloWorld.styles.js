import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
  mainView: {
    flexDirection: 'column',
    flex: 1,
    justifyContent: 'center'
  },
  helloWorld: {
    textAlign: 'center'
  }
})
