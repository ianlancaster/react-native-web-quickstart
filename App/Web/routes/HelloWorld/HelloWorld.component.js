import React from 'react'
import classes from './HelloWorld.styles.scss'

const HelloWorld = () => {
  return (
    <div className={classes.container}>
      <h2 className={classes.text}>Hello World!</h2>
    </div>
  )
}

module.exports = HelloWorld
