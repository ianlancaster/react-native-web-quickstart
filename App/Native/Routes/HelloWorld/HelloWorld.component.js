import React, { Component, PropTypes } from 'react'
import { View, Text } from 'react-native'
import { styles } from './HelloWorld.styles'

export default class HelloWorld extends Component {
  render () {
    const { toggleColor, color } = this.props

    return (
      <View style={styles.mainView}>
        <Text onPress={toggleColor} style={[styles.helloWorld, { color }]}>Hello World</Text>
      </View>
    )
  }
}

HelloWorld.propTypes = {
  color: PropTypes.string.isRequired,
  toggleColor: PropTypes.func.isRequired
}
