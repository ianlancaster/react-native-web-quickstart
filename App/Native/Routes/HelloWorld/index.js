import HelloWorld from './HelloWorld.component.js'

export default HelloWorld
