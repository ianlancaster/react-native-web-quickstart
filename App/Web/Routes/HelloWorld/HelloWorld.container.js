import { connect } from 'react-redux'
import { toggleColor } from '../../../Modules/helloWorld'
import HelloWorld from './HelloWorld.component'

const mapDispatchToProps = {
  toggleColor
}

const mapStateToProps = (state, ownProps) => ({
  color: state.helloWorld.color
})

export default connect(mapStateToProps, mapDispatchToProps)(HelloWorld)
