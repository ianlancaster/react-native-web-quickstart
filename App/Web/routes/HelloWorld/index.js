import HelloWorld from './HelloWorld.component'

export default {
  path: 'hello-world',
  component: HelloWorld
}
