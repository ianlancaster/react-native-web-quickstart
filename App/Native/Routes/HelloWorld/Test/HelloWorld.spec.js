import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import HelloWorld from '../HelloWorld.component'
import { props } from './HelloWorld.setup'

it('renders correctly', () => {
  const tree = renderer.create(
    <HelloWorld {...props} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
