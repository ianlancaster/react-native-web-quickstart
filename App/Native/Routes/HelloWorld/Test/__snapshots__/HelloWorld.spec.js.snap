// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "fontFamily": "HelveticaNeue",
          "fontSize": 20,
          "textAlign": "center",
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    Hello World!
  </Text>
</View>
`;
